int16 x
int16 y
bool rightObstacle
bool leftObstacle